jrcc-access-spring-boot-autoconfigure/src/main/java/ca/gov/bc/open/jrccaccess/autoconfigure/plugins/rabbitmq/RedisStorageService.java

```diff
@@ -55,6 +56,27 @@ public DocumentStorageProperties putString(String content) throws ServiceUnavail
 		}
 
 	}
+	
+	@Override
+	public String getString(String key, String digest) throws ServiceUnavailableException {
+
+
+		try {
+			ValueWrapper valueWrapper = this.cacheManager.getCache("Document").get(key);
+			String content = (String) valueWrapper.get();
+			String digestToCompare = computeMd5(content);
+			
+			if(digestToCompare.equals(digest)) {
+				return content;
+			}
+			
+			// TODO throw an exception
+			return null;
```

Comment:
Agree we should throw an exception if the MD5 doesn't verify, should we create a new custom exception